vci-service-impl/src/main/java/io/mosip/certify/vci/services/VCICacheService.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+package io.mosip.certify.vci.services;
+
+import io.mosip.certify.core.dto.vci.VCIssuanceTransaction;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.CachePut;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class VCICacheService {
+
+    @Autowired
+    private CacheManager cacheManager;
+
+    private static final String VCISSUANCE_CACHE = "vcissuance";
+
+    @CachePut(value = VCISSUANCE_CACHE, key = "#accessTokenHash")
+    public VCIssuanceTransaction setVCITransaction(String accessTokenHash, VCIssuanceTransaction vcIssuanceTransaction) {
+        return vcIssuanceTransaction;
+    }
+
+    public VCIssuanceTransaction getVCITransaction(String accessTokenHash) {
+        return cacheManager.getCache(VCISSUANCE_CACHE).get(accessTokenHash, VCIssuanceTransaction.class); //NOSONAR getCache() will not be returning null here.
+    }
```

Comment:
This code assumes there is a shared cache between the eSignet and Cerify. But that is not correct and we have to find a way to handle this nonce without cache.

Reply:
as of now cNounce is fetched from token every time since cache is not accessible here, while segregating mosip Id we will have a proper implementation of accessing cache which we will add it here as well

Reply:
Please create a task specifically for Nonce, so we don't miss this.

Reply:
created a task for [this](https://mosip.atlassian.net/browse/INJICERT-209)